### PR TITLE
chore(Engine): bump Node.js to v14.0.0

### PR DESCRIPTION
**Pull request by kyranet**
**Please describe the changes this PR makes and why it should be merged:**

For v13.x, we are permitted to do breaking changes, and can clean up a lot of code (specially nullability checks) with the new ES2020 syntax, specially with `?.` and `??`.

Node.js v14.0.0 is the lowest version that supports both features.

**Status**

- [ ] Code changes have been tested against the Discord API, or there are no code changes
- [ ] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [ ] This PR changes the library's interface (methods or parameters added)
  - [x] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Review by vladfrangu — APPROVED** [review]

**Comment by advaith1** [comment]
While having access to the new syntax would be great, since there are so many glitch bots we might wanna consider that glitch [currently only supports up to 12](https://nodeversions.glitch.me)

**Comment by iCrawl** [comment]
@advaith1 I certainly do not care about glitch support.

While orientating yourself around services that support our software, I am not too much inclined in completely working around them.

**Comment by kyranet** [comment]
Furthermore I wouldn't expect v13 to be released for at least a month, by that time Glitch should have support for v14, which has been LTS for a month and a half.

discord.js v12 will remain working for a few more months (until Discord kills v6 WS and/or REST, which I don't see happening until Summer 2021) if you need a version working with Node.js v12.

**Review by MattIPv4 — COMMENTED** [review]
GitHub workflows should probably also be bumped to node 14 at the same time?

**Review by Sidemen19 — APPROVED** [review]

**Review by MattIPv4 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -41,11 +41,11 @@ discord.js is a powerful [Node.js](https://nodejs.org) module that allows you to
 
 ## Installation
 
-**Node.js 12.0.0 or newer is required.**  
+**Node.js 14.0.0 or newer is required.**
```
Comment (SpaceEEC) [inline README.md:44]:
```suggestion
**Node.js 14.0.0 or newer is required.**  
```

File: README.md [hunk 2]
```diff
@@ -41,11 +41,11 @@ discord.js is a powerful [Node.js](https://nodejs.org) module that allows you to
...
 Ignore any warnings about unmet peer dependencies, as they're all optional.
 
-Without voice support: `npm install discord.js`  
-With voice support ([@discordjs/opus](https://www.npmjs.com/package/@discordjs/opus)): `npm install discord.js @discordjs/opus`  
+Without voice support: `npm install discord.js`
+With voice support ([@discordjs/opus](https://www.npmjs.com/package/@discordjs/opus)): `npm install discord.js @discordjs/opus`
```
Comment (SpaceEEC) [inline README.md:48]:
```suggestion
Without voice support: `npm install discord.js`  
With voice support ([@discordjs/opus](https://www.npmjs.com/package/@discordjs/opus)): `npm install discord.js @discordjs/opus`  
```

File: README.md [hunk 3]
```diff
@@ -102,7 +102,7 @@ client.login('token');
 ## Contributing
 
 Before creating an issue, please ensure that it hasn't already been reported/suggested, and double-check the
-[documentation](https://discord.js.org/#/docs).  
+[documentation](https://discord.js.org/#/docs).
```
Comment (SpaceEEC) [inline README.md:105]:
```suggestion
[documentation](https://discord.js.org/#/docs).  
```

File: docs/general/welcome.md [hunk 4]
```diff
@@ -33,11 +33,11 @@ discord.js is a powerful [Node.js](https://nodejs.org) module that allows you to
 
 ## Installation
 
-**Node.js 12.0.0 or newer is required.**  
+**Node.js 14.0.0 or newer is required.**
```
Comment (SpaceEEC) [inline docs/general/welcome.md:36]:
```suggestion
**Node.js 14.0.0 or newer is required.**  
```

File: docs/general/welcome.md [hunk 5]
```diff
@@ -33,11 +33,11 @@ discord.js is a powerful [Node.js](https://nodejs.org) module that allows you to
...
 Ignore any warnings about unmet peer dependencies, as they're all optional.
 
-Without voice support: `npm install discord.js`  
-With voice support ([@discordjs/opus](https://www.npmjs.com/package/@discordjs/opus)): `npm install discord.js @discordjs/opus`  
+Without voice support: `npm install discord.js`
+With voice support ([@discordjs/opus](https://www.npmjs.com/package/@discordjs/opus)): `npm install discord.js @discordjs/opus`
```
Comment (SpaceEEC) [inline docs/general/welcome.md:40]:
```suggestion
Without voice support: `npm install discord.js`  
With voice support ([@discordjs/opus](https://www.npmjs.com/package/@discordjs/opus)): `npm install discord.js @discordjs/opus`  
```

File: docs/general/welcome.md [hunk 6]
```diff
@@ -94,7 +94,7 @@ client.login('token');
 ## Contributing
 
 Before creating an issue, please ensure that it hasn't already been reported/suggested, and double-check the
-[documentation](https://discord.js.org/#/docs).  
+[documentation](https://discord.js.org/#/docs).
```
Comment (SpaceEEC) [inline docs/general/welcome.md:97]:
```suggestion
[documentation](https://discord.js.org/#/docs).  
```

**Review by SpaceEEC — CHANGES_REQUESTED** [review]
You seem to have dropped a few significant spaces here.

**Comment by Vaporox** [comment]
Should this also be updated?
https://github.com/discordjs/discord.js/blob/master/docs/general/faq.md#L7

**Comment by advaith1** [comment]
this pr already updates that

**Comment by Vaporox** [comment]
oops yeah, mb

**Review by iCrawl — APPROVED** [review]

**Pull request merged**